grails-app/views/order/_orderInvoices.gsp

```diff
@@ -19,7 +19,14 @@
                     <tr class="${i%2?'even':'odd'}">
                         <td>${invoiceItem?.orderItem?.id}</td>
                         <td>${invoiceItem?.product?.productCode}</td>
-                        <td>${invoiceItem?.description}</td>
```

Comment:
@EWaterman I'm sorry, I forgot to finish the answer yesterday.
 `description` is defined like this:
```groovy
String getDescription() {
        return orderAdjustment ? orderAdjustment.description : product?.name
}
```
so on the frontend side I had to find some way to determine whether I return the `orderAdjustment`'s description or the product name. If the product name, then we'd want to return a synonym of the product, otherwise we don't want to change anything and still return the `orderAdjustment.description`.

Reply:
> Yeah, feels like we could try to move this to the domain (and return either description or product)

This is currently in the domain, so I might not understand what you mean.
If you meant a different method, still, it would be impossible to distinguish on the frontend side (without having the if logic) if it's the description or the product.
In my opinion trying to handle this one, minor case in the generic `format:displayName` seems not worth it, unless we face more of those edge cases.

Reply:
You are currently returning a product name in the domain (unless it somehow returns translated one)

Reply:
yes it does, but I didn't want to change it to return the translated name (in case somewhere it is not expected, and the `description` transient is used) + again, on the frontend I would still need a mechanism to determine whether it is the product name, or the order adjustment description in order to know, if I should display the tooltip (we used to add tooltips with an original name of a product).

Reply:
That's why I wrote: "It feels like we could try to move this to the domain (...)", but we'd need to also "extend the displayName tag", but it might unnecessarily overcomplicate it and does not 100% feel right.